### PR TITLE
Fixed#2922

### LINKED ISSUE #2922: email to cases failure with email body > 65536 Bytes
suitecrm 7.7.9
email to cases failure with email body > 65536 Bytes

the email body does not fit inside mysql description TEXT column causing a mysql error when case is created. 

solution: email body to be truncated before mysql query.

### PR DESCRIPTION
**Pull request by boombata**
Fixed #2922 - email to cases failure with email body > 65536 Bytes
## Description

Changes the cases vardef to allow the description field of a case to be greater than 65536 Bytes.

Issue ref: #2922
this pr in replacement of https://github.com/salesagility/SuiteCRM/pull/4334 because is a side effect doing changing only vardef, the detail , edit and quick create views do not display description field as a multi lines field but just single line.

## Motivation and Context

  This fixes a case failing to create when using the quick create from the emails module. This was due to an SQL insert query failure when the text of the email was above 65536 Bytes.

## How To Test This

  Create an email with 65536 bytes of text.
    Click on quick create > case

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

File: modules/DynamicFields/templates/Fields/TemplateTextLongArea.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	var $type = 'longtext';
+	var $len = '';
+
+	function __construct()
+	{
+		$this->vardef_map['rows'] = 'ext2';
+		$this->vardef_map['cols'] = 'ext3';
+	}
+
+	function set($values){
```
Comment (PedroErnst) [inline modules/DynamicFields/templates/Fields/TemplateTextLongArea.php]: Method visibility should be explicitly declared (private / public). Also the methods missing phpdocs.

**Comment by boombata** [comment]
@PedroErnst should be ok now. thanks

**Comment by Mac-Rae** [comment]
Please resolve the conflicts on this pull request bringing it up to date.

**Comment by pgorod** [comment]
I've been playing with textareas and TinyMCE editors recently. I find that the code is a bit of a mess with different field types, and often the implications of moving away from one field type to another are not always completely understood.

We have a user-profile setting for the kind of editor to use (TinyMCE, Mozaik, Direct HTML) that people often complain about, because it affects some rich text editors throughout the product, but not others.
 
My idea for all this was to use a new field type called **editor** which would allow for all those 3 editors (and any future ones) to be used seamlessly in any module/view.

The multiplication of field types is not good for this. 

I think the current problem should be solved by using a **db-type** to force the long text on the database size, but without forking the **`modules/DynamicFields/templates/Fields/TemplateTextArea.php`** file, repeating code. The field type would stay the same as before, only the database would use a longer field.

Sorry for not saying this sooner, but I simply didn't understand this sooner...

**Pull request closed without merging**